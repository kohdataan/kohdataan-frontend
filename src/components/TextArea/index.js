import React, { memo } from 'react'
import PropTypes from 'prop-types'

const TextArea = props => {
  const { label, value, inputClassName, labelClassName, onChange, rows } = props
  return (
    <label htmlFor={label} className={inputClassName}>
      {label}
      <textarea
        name={label}
        id={label}
        rows={rows}
        value={value}
        placeholder={label}
        className={labelClassName}
        onChange={onChange}
      />
    </label>
  )
}

TextArea.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string,
  inputClassName: PropTypes.string,
  labelClassName: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  rows: PropTypes.string,
}

TextArea.defaultProps = {
  value: '',
  inputClassName: '',
  labelClassName: '',
  rows: '3',
}

export default memo(TextArea)
