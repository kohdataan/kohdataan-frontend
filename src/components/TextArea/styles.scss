
::-webkit-resizer {
    display: none;
  }

textarea {
  resize: none;
}