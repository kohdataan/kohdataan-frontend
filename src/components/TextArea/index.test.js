import React from 'react'
import { create } from 'react-test-renderer'
import TextArea from './index'

describe('TextArea', () => {
  test('it matches the snapshot', () => {
    const component = create(
      <TextArea label="test" onChange={() => {}}>
        <p>test</p>
      </TextArea>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
