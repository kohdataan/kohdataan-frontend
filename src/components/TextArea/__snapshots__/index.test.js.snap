// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea it matches the snapshot 1`] = `
<label
  className=""
  htmlFor="test"
>
  test
  <textarea
    aria-describedby={null}
    aria-label="test"
    className=""
    id="test"
    maxLength={1000}
    name="test"
    onChange={[Function]}
    placeholder="test"
    rows={4}
    type="text"
    value=""
  />
</label>
`;
